src/index.ts

```diff
@@ -24,127 +24,169 @@ export type StorageArea = browser.Storage.StorageArea &
 
 export type InternalStorage = browser.Storage.Static
 
-const hasWindow = typeof window !== "undefined"
-/**
- * https://docs.plasmo.com/framework/storage
- */
-export class Storage {
-  #storage: InternalStorage
-  #client: StorageArea
-  #localClient = hasWindow ? window.localStorage : null
+const hasWebApi = typeof window !== "undefined"
+
+export abstract class BaseStorage {
+  #extStorageEngine: InternalStorage
+
+  // The main storage engine
+  #extClient: StorageArea
+  get extClient() {
+    return this.#extClient
+  }
+
+  // The backup client on the web, for now use localStorage
+  #webClient = hasWebApi ? window.localStorage : null
+  get webClient() {
+    return this.#webClient
+  }
 
   #area: StorageAreaName
-  #skipQuotaCheck = false
+  get area() {
+    return this.#area
+  }
 
-  // TODO: Make another map for local storage
-  #chromeStorageCommsMap: Map<
+  get hasWebAPI() {
+    return hasWebApi
+  }
+
+  #shouldCheckQuota = false
+
+  #chromeStorageCommsMap = new Map<
     string,
     {
       callbackSet: Set<StorageWatchCallback>
       listener: StorageWatchEventListener
     }
-  > = new Map()
+  >()
+
+  #publicKeySet: Set<string>
+  get publicKeySet() {
+    return this.#publicKeySet
+  }
 
-  #secretSet: Set<string>
-  #allSecret = false
+  isPublic = (key: string) => this.allPublic || this.publicKeySet.has(key)
```

Comment:
same for all other uses of public

Reply:
Aight renamed it to "copied" since I want this to be even more generic down the line (rn it sync between web - ext. But the interface can actually be used for web - backend or ext - backend storage too)